packages/@aws-cdk/core/lib/cfn-mapping.ts

```diff
@@ -123,3 +126,41 @@ export class CfnMapping extends CfnRefElement {
     }
   }
 }
+
+class CfnMappingEmbedder implements IResolvable {
+  readonly creationStack: string[] = [];
+
+  constructor(private readonly mappingId: string, private readonly mapping: Mapping, private readonly value: string) { }
+
+  private tryFindChildRecurse(scope: Construct): boolean {
+    if (scope.node.tryFindChild(this.mappingId)) {
+      return true;
+    }
+
+    for (const child of scope.node.children) {
+      if (this.tryFindChildRecurse(child)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  public resolve(context: IResolveContext): string {
+    const consumingStack = Stack.of(context.scope);
+    // we must check the entire tree because CfnMappings can be
+    // added to Constructs and won't be direct children of the
+    // stack. This is done in aws-secretsmanager/lib/secret-rotation.ts
```

Comment:
We can make it slightly simpler though.

If `Stack.of(context.scope) === Stack.of(cfnMapping)`, we don't need to do anything anyway -- so we don't care where in the tree it is!

And if we copy it to another stack, we can decide where to put it (directly under the Stack would be an obvious place).

Reply:
I like the idea, but this doesn't quite work; if `resolve()` is called more than once in that context, this check will not block us from trying to create the mapping again.